File: WebAPI2/StoryBoard/Dto/TaskDto.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Dto;

public class TaskDto
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime CreationDate { get; set; }
    public DateTime EndDate { get; set; }
    public int DurationTime { get; set; }
    public string Status { get; set; }
    public int IdStory { get; set; }
}

File: UI/StoryBoard/src/app/dtos/task-dto/task-dto.ts
export class TaskDto {
    id : number = 0;
    name : string = "";
    description : string = "";
    creationDate : Date = new Date;
    endDate : Date = new Date;
    durationTime : number = 0;
    status : string = "";
    idStory: number = 0;

    static createFromFormValues(formValues: any): TaskDto {
        const task = new TaskDto();
        task.id = formValues.id;
        task.name = formValues.name;
        task.description = formValues.description;
        task.creationDate = formValues.creationDate;
        // task.endDate = formValues.endDate;
        task.durationTime = formValues.durationTime;
        task.status = formValues.status;
        task.idStory = formValues.idStory;
        return task;
    }
}